pints/_log_likelihoods.py

```diff
@@ -117,6 +117,120 @@ def __call__(self, x):
                       - np.sum(autocorr_error**2, axis=0) / (2 * sigma**2))
 
 
+class GaussianIntegratedUniformLogLikelihood(pints.ProblemLogLikelihood):
+    """
+    Calculates a log-likelihood assuming independent Gaussian-distributed noise
+    at each time point where it is assumed that the dependence on
+    :math:`\\sigma\\sim U(a,b)` has been integrated out of the joint posterior.
+
+    The likelihood is given in terms of the sum of squared errors:
+
+    .. math::
+        SSE = \\sum_{i=1}^n (f_i(\\theta) - y_i)^2
+
+    and is given up to a normalisation constant by:
+
+    .. math::
+        \\text{log } L = - n / 2 \\text{log}(\\pi) -
+            \\text{log}(2 (b - a) \\sqrt(2)) +
+            (1 / 2 - n / 2) \\text{log}(SSE) +
+            \\text{log}\\left[\\Gamma((n - 1) / 2, \\frac{SSE}{2 b^2}) -
+            \\Gamma((n - 1) / 2, \\frac{SSE}{2 a^2}) \\right]
+
+    where :math:`\\Gamma(u,v)` is the upper incomplete gamma function
+    as defined here: https://en.wikipedia.org/wiki/Incomplete_gamma_function
+
+    Arguments:
+
+    ``problem``
+        A :class:`SingleOutputProblem` or :class:`MultiOutputProblem`.
+    ``b``
+        The upper limit on the uniform prior om `sigma`.
+
+    *Extends:* :class:`ProblemLogLikelihood`
+    """
+
+    def __init__(self, problem, lower, upper):
+        super(GaussianIntegratedUniformLogLikelihood, self).__init__(problem)
+
+        # Get number of times, number of outputs
+        self._nt = len(self._times)
+        self._no = problem.n_outputs()
+
+        # Add parameters to problem
```

Comment:
Does this method add new parameters to be inferred? If so, it should overwrite the `n_parameters()` method and return the updated value

Reply:
Nope. No new parameters are returned. See the example notebook or docstrings.

Reply:
What does that comment mean then?